docs/how-to/virtual_memory.rst

```diff
@@ -0,0 +1,102 @@
+
+.. code-block:: cpp
+
+    size_t granularity = 0;
+    hipMemGenericAllocationHandle_t allocHandle;
+    hipMemAllocationProp prop = {};
+    prop.type = HIP_MEM_ALLOCATION_TYPE_PINNED;
+    prop.location.type = HIP_MEM_LOCATION_TYPE_DEVICE;
+    prop.location.id = currentDev;
+    hipMemGetAllocationGranularity(&granularity, &prop, HIP_MEM_ALLOC_GRANULARITY_MINIMUM);
+    padded_size = ROUND_UP(size, granularity);
+    hipMemCreate(&allocHandle, padded_size, &prop, 0);
+
+Reserve Virtual Address Range
+-----------------------------
+
+After you have acquired an allocation of physical memory, you must map it before you can use it. To do so, you need a virtual address to map it to.  Mapping means the physical memory allocation is available from the virtual address range it is mapped to. To reserve a virtual memory range, use the ``hipMemAddressReserve`` function. The size of the virtual memory must match the amount of physical memory previously allocated. You can then map the physical memory allocation to the newly-acquired virtual memory address range using the ``hipMemMap`` function.
+
+.. code-block:: cpp
+
+    hipMemAddressReserve(&ptr, padded_size, 0, 0, 0);
+    hipMemMap(ptr, padded_size, 0, allocHandle, 0);
+
+Set Memory Access
+-----------------
+
+Finally, use the ``hipMemSetAccess`` function to enable memory access. It accepts the pointer to the virtual memory, the size, and a ``hipMemAccessDesc`` descriptor as parameters.
+
+.. code-block:: cpp
+
+    hipMemAccessDesc accessDesc = {};
+    accessDesc.location.type = HIP_MEM_LOCATION_TYPE_DEVICE;
+    accessDesc.location.id = currentDev;
+    accessDesc.flags = HIP_MEM_ACCESS_FLAGS_PROT_READWRITE;
+    hipMemSetAccess(ptr, padded_size, &accessDesc, 1);
+
+At this point the memory is allocated, mapped, and ready for use. You can read and write to it, just like you would a C style memory allocation.
+
+Free Virtual Memory
+-------------------
+
+To free the memory allocated in this manner, use the corresponding free functions. To unmap the memory, use ``hipMemUnmap``. To release the virtual address range, use ``hipMemAddressFree``.  Finally, to release the physical memory, use ``hipMemRelease``.
+
+.. code-block:: cpp
+
+    hipMemUnmap(ptr, size);
+    hipMemRelease(allocHandle);
+    hipMemAddressFree(ptr, size);
+
+.. _usage_virtual_memory:
+Usage
+=====
+
+Increase Allocation Size
```

Comment:
Should this be _Dynamically increase allocation size_? It seems this is the crux of the reason for using virtual memory, isn't it?